### PR TITLE
[v17] docs: update cloud networking on proxy service

### PR DESCRIPTION
**Pull request by stevenGravy**
Backport #47506 to branch/v17

### CODE REVIEW

File: docs/pages/reference/networking.mdx [hunk 1]
```diff
@@ -36,21 +36,9 @@ following use cases:
...
-All Teleport services (e.g., the Application Service and Database Service) have
-an optional `public_addr` property that you can modify in each service's
-configuration file. The public address can take an IP or a DNS name. It can also
-be a list of values:
-
-```yaml
-public_addr: ["service-one.example.com", "service-two.example.com"]
-```
+On Teleport Enterprise (Cloud) the Teleport agent services always
+connect using reverse tunnels so there is no need to set a public address for a agent.
```
Comment (mmcallister) [inline docs/pages/reference/networking.mdx:40]:
```suggestion
connect using reverse tunnels so there is no need to set a public address for an agent.
```

**Comment by stevenGravy** [comment]
good to approve this and the other backports @ptgott @mmcallister ?

File: docs/pages/reference/networking.mdx [hunk 2]
```diff
@@ -36,21 +36,9 @@ following use cases:
...
 
-All Teleport services (e.g., the Application Service and Database Service) have
-an optional `public_addr` property that you can modify in each service's
-configuration file. The public address can take an IP or a DNS name. It can also
-be a list of values:
-
-```yaml
-public_addr: ["service-one.example.com", "service-two.example.com"]
-```
+On Teleport Enterprise (Cloud) the Teleport agent services always
```
Comment (mmcallister) [inline docs/pages/reference/networking.mdx:39]:
```suggestion
On Teleport Enterprise (Cloud) the Teleport Agent services always
```

**Review by mmcallister — APPROVED** [review]

**Review by mmcallister — APPROVED** [review]

**Review by scottpgallagher — APPROVED** [review]
Approved.

**Pull request merged**